frontend/src/metabase/admin/settings/components/ApiKeys/CreateApiKeyModal.tsx

```diff
@@ -0,0 +1,81 @@
+import { t } from "ttag";
+import { useCallback, useState } from "react";
+
+import { Text, Button, Group, Modal, Stack } from "metabase/ui";
+import {
+  Form,
+  FormErrorMessage,
+  FormProvider,
+  FormGroupWidget,
+  FormSubmitButton,
+  FormTextInput,
+} from "metabase/forms";
+import { ApiKeysApi } from "metabase/services";
+
+import { SecretKeyModal } from "./SecretKeyModal";
+
+export const CreateApiKeyModal = ({
+  onClose,
+  refreshList,
+}: {
+  onClose: () => void;
+  refreshList: () => void;
+}) => {
+  const [modal, setModal] = useState<"create" | "secretKey">("create");
+  const [secretKey, setSecretKey] = useState<string>("");
+
+  const handleSubmit = useCallback(
+    async vals => {
+      const response = await ApiKeysApi.create(vals);
+      setSecretKey(response.unmasked_key);
+      setModal("secretKey");
+      refreshList();
+    },
+    [refreshList],
+  );
+
+  if (modal === "secretKey") {
+    return <SecretKeyModal secretKey={secretKey} onClose={onClose} />;
+  }
+
+  if (modal === "create") {
+    return (
+      <Modal
+        size="30rem"
+        padding="xl"
+        opened
+        onClose={onClose}
+        title={t`Create a new API Key`}
+      >
+        <FormProvider initialValues={{}} onSubmit={handleSubmit}>
+          <Form data-testid="create-api-key-modal">
+            <Stack spacing="md">
+              <FormTextInput
+                name="name"
+                label={t`Key name`}
+                size="sm"
+                required
+              />
```

Comment:
I'm not sure this is important, but it seemed worth mentioning, just in case. If you enter nothing in this field, the form shows the error "value must be a non-blank string." But if you enter a very, very long string, you get an error message with SQL in it, like this:

<img width="300" alt="image" src="https://github.com/metabase/metabase/assets/130925/d6b384ea-3793-4f29-b184-458567d66629">

Reply:
Here's another slightly unexpected bit of behavior in this modal.

If I re-use an API key name, I get this normal-looking error message:

<img width="300" alt="image" src="https://github.com/metabase/metabase/assets/130925/a7be86a4-8dc8-4dd0-9f1c-e31ac124d811">

However, when I blur the Key name text field, the phrase 'An error occurred' pops up:

<img width="300" alt="image" src="https://github.com/metabase/metabase/assets/130925/d5693539-67d4-4574-824b-d62f0ccabbda">

Reply:
The following issue seems like it might be a bit more serious.

If I create a key named "a" and another key named "a " (with a space), that works. But if I then delete both keys and try to create a key named "a", I get this error:

<img width="400" alt="image" src="https://github.com/metabase/metabase/assets/130925/b09123b8-583e-46e9-a787-a6ec4d6bae1f">

I think I've now put my database into a state where it is impossible to make a key named "a".

Reply:
Another small issue relating to key names. Since non-alphanumeric characters are stripped from the key name ([here](https://github.com/metabase/metabase/blob/4d5b621879173c187d28bb58b234197d31dfbe7e/src/metabase/api/api_key.clj#L55)) to make an email address for internal use, if you make a key named `$a` and try to make a second key named `@a`, you get an error:

<img width="500" alt="image" src="https://github.com/metabase/metabase/assets/130925/057b27f7-80a0-4703-86da-12dc404ebc4f">

If this is a genuine problem, maybe we can append a timestamp before the '@' in the email address, in the backend?

Reply:
I think there might be a backend issue that isn't deleting users when API keys are deleted? @noahmoss (John is out this week) do you know if that is working as intended?

Reply:
If I recall correctly, it was a conscious decision not to delete the users, since they could be the owners of existing content and we'd still want their names to display correctly on the UI.

It sounds like there are some bugs here though. We need to make sure that emails are unique, even when API Key names are the same as previously-deleted ones. 

Could you file an issue (or issues) and assign them to John for when he's back?

Reply:
I'm not too worried about the length errors (though I did add a max length to the input). what is scary is what having a 250 char key name does to the table formatting 😱

will constrain it.